eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/Utils.java

```diff
@@ -44,15 +46,38 @@
     public static final String AZURE_PERF_LIBRARY_IDENTIFIER = "-perf";
     public static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();
     public static final Pattern STRING_SPLIT_BY_DOT = Pattern.compile("[.]");
+    public static final String PROJECT_VERSION = "project.version";
 
     public static final HashSet<String> RESOLVED_EXCLUSION_LIST = new HashSet<>(Arrays.asList(
        "junit-jupiter-api"
     ));
 
+    public static final HashSet<String> IGNORE_CONFLICT_LIST = new HashSet<>(Arrays.asList(
+        "slf4j-api" // slf4j is compatible across versions.
+    ));
+
     public static final String POM_TYPE = "pom";
     private static Logger logger = LoggerFactory.getLogger(Utils.class);
 
-    public static List<BomDependency> getExternalDependenciesContent(List<Dependency> dependencies) {
+    static void validateNotNullOrEmpty(String argValue, String argName) {
+        if(argValue == null || argValue.isEmpty()) {
+            throw new NullPointerException(String.format("%s can't be null", argName));
+        }
+    }
+
+    static void validateNull(String argValue, String argName) {
+        if(argValue != null) {
+            throw new IllegalArgumentException(String.format("%s should be null", argName));
+        }
+    }
+
+    static void validateValues(String argName, String argValue, String ... expectedValues) {
+        if(Arrays.stream(expectedValues).noneMatch(a -> a.equals(argValue))) {
+            throw new IllegalArgumentException(String.format("%s must match %s", argName, expectedValues));
```

Comment:
nit: `String.format("%s", array)` will print the first value in the array. It won't print all the values in the array. Given the error message here, I am assuming that this is intended to print all expected values.

Reply:
Thank you :)